tests/commands/harddrive.py

```diff
@@ -75,5 +80,36 @@ def runTest(self):
         # unknown option
         self.assert_parse_error("harddrive --unknown=value")
 
+
+class F33HardDrive_TestCase(unittest.TestCase):
+    def runTest(self):
+        data1 = F33_HardDrive()
+        data2 = F33_HardDrive()
+
+        attribute_test(data1, data2, self)
+
+
+class F33_TestCase(CommandTest):
+    def runTest(self):
+        # pass
+        self.assert_parse("harddrive --dir=/install --partition=part", "harddrive --dir=/install --partition=part\n")
+        self.assertFalse(self.assert_parse("harddrive --dir=/install --partition=sda1") is None)
+        self.assertTrue(self.assert_parse("harddrive --dir=/install --partition=sda1") !=
+                        self.assert_parse("harddrive --dir=/install --partition=sda2"))
+
+        # fail
+        # Ensure these options have been removed.
+        self.assert_removed("harddrive", "--biospart")
+        # required option --dir missing
+        self.assert_parse_error("harddrive")
+        # required --dir argument missing
+        self.assert_parse_error("harddrive --dir")
+        # missing --biospart or --partition option
+        self.assert_parse_error("harddrive --dir=/install")
+        # both --biospart and --partition specified
+        self.assert_parse_error("harddrive --dir=/install --partition")
+        # unknown option
+        self.assert_parse_error("harddrive --unknown=value")
+
```

Comment:
Add checks for `harddrive --partition` and `harddrive --partition=/install` since both `--partition` and `--dir` are now required parameters.

Reply:
Fixed if I understand it correctly that the second should be ``... --dir=/install``.